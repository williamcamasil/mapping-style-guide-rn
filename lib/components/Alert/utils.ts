import { ViewStyle } from 'react-native';

import { Icons } from '../../assets';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type AlertVariantType =
  'success' |
  'info' |
  'warning' |
  'error';

export const getVariantColorStyle = (variant?: AlertVariantType): ColorPaletesType => {
  switch (variant) {
    case 'warning':
      return 'feedbackAlert500';
    case 'error':
      return 'feedbackError500';
    case 'info':
      return 'neutralGray600';
    case 'success':
    default:
      return 'feedbackSuccess500';
  }
};

export const getDefaultIcon = (variant?: AlertVariantType) => {
  switch (variant) {
    case 'success':
      return Icons.Default.Success;
    case 'info':
      return Icons.Default.Information;
    case 'warning':
    case 'error':
    default:
      return Icons.Default.Warning;
  }
};

export const getAlertStyle = (
  theme: AppThemeType,
  variant?: AlertVariantType,
): ViewStyle => {

  switch (variant) {
    case 'warning':
      return {
        backgroundColor: theme.colors.feedbackAlert100,
      };
    case 'error':
      return {
        backgroundColor: theme.colors.feedbackError100,
      };
    case 'info':
      return {
        backgroundColor: theme.colors.neutralGray100,
      };
    case 'success':
    default:
      return {
        backgroundColor: theme.colors.feedbackSuccess100,
      };
  }
};
