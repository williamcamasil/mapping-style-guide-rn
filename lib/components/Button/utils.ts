import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
