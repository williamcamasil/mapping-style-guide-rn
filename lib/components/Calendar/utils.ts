import { dateToString } from 'mapping-context-rn';
import type { MarkedDates, Theme } from 'react-native-calendars/src/types';

import { AppThemeType } from '../../theme';

type ExtendedTheme = Theme & {
  'stylesheet.calendar.header': object;
  'stylesheet.day.basic': object;
};

/**
 * Define o tipo de data para o formato yyyy-MM-dd
 */
export type StringDateType = `${number}${number}${number}${number}-${number}${number}-${number}${number}`;

export const DATE_FORMAT = 'yyyy-MM-dd';

export const DAY_SIZE = 42;

export const getDefaultValue = (value?: StringDateType | null) => (
  value ?? dateToString(new Date(), DATE_FORMAT) as StringDateType
);

export const getDisabledDates = (disabledDates?: StringDateType[]) => {
  if (!disabledDates?.length) return {};
  const dates: MarkedDates = {};
  disabledDates.forEach((date: StringDateType) => {
    dates[date] = {
      disabled: true,
      disableTouchEvent: true,
    };
  });
  return dates;
};

export const getCommonTheme = (theme: AppThemeType): ExtendedTheme => ({
  selectedDayBackgroundColor: theme.colors.primaryMain,
  todayTextColor: theme.colors.primaryMain,
  dayTextColor: theme.colors.neutralGray600,
  textDayFontFamily: theme.typography.weights.bold.fontFamily,
  textMonthFontFamily: theme.typography.weights.bold.fontFamily,
  textDayHeaderFontFamily: theme.typography.weights.bold.fontFamily,
  textDayFontWeight: theme.typography.weights.bold.fontWeight,
  textMonthFontWeight: theme.typography.weights.bold.fontWeight,
  textDayHeaderFontWeight: theme.typography.weights.bold.fontWeight,
  textDayFontSize: theme.typography.sizes.small,
  textMonthFontSize: theme.typography.sizes.small,
  textDayHeaderFontSize: theme.typography.sizes.small,
  'stylesheet.calendar.header': {
    dayHeader: {
      marginTop: theme.spacings.sXS,
      marginBottom: theme.spacings.sXS,
      textAlign: 'center',
      fontSize: theme.typography.sizes.small,
      color: theme.colors.primaryMain,
      fontFamily: theme.typography.weights.bold.fontFamily,
      fontWeight: theme.typography.weights.bold.fontWeight,
      flex: 1,
    },
    week: {
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  'stylesheet.day.basic': {
    base: {
      height: DAY_SIZE,
      width: DAY_SIZE,
      borderBottomLeftRadius: theme.borders.radius.XL,
      borderBottomRightRadius: theme.borders.radius.XL,
      borderTopLeftRadius: theme.borders.radius.XL,
      borderTopRightRadius: theme.borders.radius.XL,
      alignItems: 'center',
      justifyContent: 'center',
    },
  },
});
