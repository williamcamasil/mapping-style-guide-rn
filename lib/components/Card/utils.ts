import { ColorValue, ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';

export type CardSizeType =
  'small' |
  'big';

export type CardVariantType =
  'default' |
  'highlighted50' |
  'highlighted100';

export const getCardSizeStyle = (theme: AppThemeType, size?: CardSizeType): ViewStyle => {
  switch (size) {
    case 'small':
      return {
        width: 135,
        height: 120,
        padding: theme.spacings.sXS,
      };
    case 'big':
    default:
      return {
        minHeight: 128,
        padding: theme.spacings.sSmall,
      };
  }
};

export const getCardShadowStyle = (theme: AppThemeType, showShadow?: boolean): ViewStyle | null => {
  if (showShadow) {
    return theme.shadows.level4;
  }

  return {
    borderWidth: theme.borders.width.thin,
    borderColor: theme.colors.neutralGray200,
  };
};

export const getCardVariantStyle = (theme: AppThemeType, cardType?: CardVariantType): ViewStyle | null => {
  switch (cardType) {
    case 'highlighted50':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary700,
      };
    case 'highlighted100':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary600,
        backgroundColor: theme.colors.primaryMain,
      };
    case 'default':
    default:
      return null;
  }
};

export const getCardCustomBorderStyle = (theme: AppThemeType, borderColor?: ColorValue): ViewStyle | null => {
  if (!borderColor) return null;

  return {
    borderWidth: theme.borders.width.thin,
    borderColor,
  };
};

export const getArrowColor = (theme: AppThemeType, variant?: CardVariantType, arrowColor?: ColorValue): ColorValue => {
  if (arrowColor) return arrowColor;

  if (variant === 'highlighted100') return theme.colors.neutralWhite;

  return theme.colors.neutralGray500;
};
