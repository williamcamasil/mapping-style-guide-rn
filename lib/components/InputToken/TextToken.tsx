import React, { useCallback } from 'react';
import { StyleSheet, View } from 'react-native';

import { useTextStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import Text from '../Text';
import {
  getInputBorderColor,
  INPUT_TOKEN_HEIGHT,
  isCurrentEmptyInputFocused,
} from './utils';

type TextTokenPropsType = {
  theme: AppThemeType;
  index: number;
  value: string;
  focused: boolean;
  maxLength: number;
  error?: boolean;
};

const styles = StyleSheet.create({
  containerText: {
    textAlign: 'center',
  },
});

const TextToken: React.FC<TextTokenPropsType> = ({
  theme, index, value, focused, error, maxLength,
}) => {

  const char = value[index];
  const enteredNumbers = value.length;

  const INPUT_TOKEN_WIDTH = maxLength > 4 ? 44 : 48;

  const textVisibleStyles = useTextStyles(() => [
    {
      borderWidth: theme.borders.width.thin,
      borderColor: getInputBorderColor(index, enteredNumbers, focused, theme, error, char),
      width: INPUT_TOKEN_WIDTH,
      height: INPUT_TOKEN_HEIGHT,
      borderRadius: theme.borders.radius.medium,
      alignItems: 'center',
      paddingTop: theme.spacings.sXS,
    },
  ], [theme, index, enteredNumbers, focused, error, char, INPUT_TOKEN_WIDTH]);

  const getInputIndexContent = useCallback(() => {
    if (char) return char;
    if (isCurrentEmptyInputFocused(focused, enteredNumbers, index)) return '|';
    return '';
  }, [char, enteredNumbers, focused, index]);

  return (
    <View
      testID={`text-token-touchable-${index}`}
      pointerEvents="none"
      style={textVisibleStyles}
    >
      <Text
        variant="subtitle"
        weight="semiBold"
        style={styles.containerText}
      >
        {getInputIndexContent()}
      </Text>
    </View>
  );
};

export default withTheme(TextToken);
