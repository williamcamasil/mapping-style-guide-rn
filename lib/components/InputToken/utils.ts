import { AppThemeType } from '../../theme/default';

export const INPUT_TOKEN_HEIGHT = 64;

export const isCurrentEmptyInputFocused = (
  focused: boolean,
  enteredNumbers: number,
  index: number,
) => focused && enteredNumbers === index;

export const getInputBorderColor = (
  index: number,
  enteredNumbers: number,
  focused: boolean,
  theme: AppThemeType,
  error?: boolean,
  value?: string,
) => {
  if (error) return theme.colors.feedbackError400;
  if (value) return theme.colors.neutralGray300;
  if (isCurrentEmptyInputFocused(focused, enteredNumbers, index)) return theme.colors.neutralGray400;
  return theme.colors.neutralGray300;
};
