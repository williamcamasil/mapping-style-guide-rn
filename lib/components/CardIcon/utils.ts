import { ColorValue } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { CardVariantType } from '../Card/utils';

export const getIconColor = (
  theme: AppThemeType,
  customColor?: ColorValue,
  variant?: CardVariantType,
): ColorValue | string => {
  if (customColor) return customColor;

  if (variant === 'highlighted100') {
    return theme.colors.neutralWhite;
  }

  return theme.colors.primaryMain;
};
