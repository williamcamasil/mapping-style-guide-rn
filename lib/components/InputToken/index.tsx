import React, {
  useCallback, useMemo, useState,
} from 'react';
import {
  NativeSyntheticEvent,
  StyleProp,
  StyleSheet,
  TextInput,
  TextInputFocusEventData,
  TextInputProps,
  View,
  ViewStyle,
} from 'react-native';

import { useTextStyles, useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { getFieldErrorState, wrapField } from '../Form';
import Spacer from '../Spacer';
import TextToken from './TextToken';
import {
  INPUT_TOKEN_HEIGHT,
} from './utils';

export type InputTokenPropsType = Omit<TextInputProps, 'placeholder' | 'onChange' | 'onChangeText' | 'editable' | 'pointerEvents' | 'maxLength'> & {
  theme: AppThemeType;
  value: string | null;
  onChange?: TextInputProps['onChangeText'];
  keyboardType?: TextInputProps['keyboardType'];
  maxLength: number;
  innerRef?: React.LegacyRef<TextInput>;
  error?: boolean;
  contentContainerStyle?: StyleProp<ViewStyle>;
};

const styles = StyleSheet.create({
  containerInput: {
    left: 0,
    right: 0,
    position: 'absolute',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  contentInputs: {
    flexDirection: 'row',
  },
});

const InputToken = ({
  theme,
  value,
  onChange,
  testID,
  innerRef,
  onFocus,
  onBlur,
  error,
  contentContainerStyle,
  ...others
}: InputTokenPropsType): React.ReactElement => {
  const [focused, setFocused] = useState(false);

  const inputStyles = useTextStyles(() => [
    {
      height: INPUT_TOKEN_HEIGHT,
      opacity: theme.opacities.transparent,
    },
  ], [theme]);

  const contentContainerStyles = useViewStyles(() => [
    styles.containerInput,
    contentContainerStyle,
  ], [contentContainerStyle]);

  const handleFocus = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(true);
      onFocus?.(e);
    },
    [onFocus],
  );

  const handleBlur = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(false);
      onBlur?.(e);
    },
    [onBlur],
  );

  const listPositions = useMemo(() => [...Array(others.maxLength)].map((_, index) => ({ index })), [others.maxLength]);

  const renderTextsToken = useCallback(() => listPositions.map(item => {
    const lastIndex = listPositions.length - 1;
    const isLastItem = item.index === lastIndex;
    return (
      <View key={item.index} style={styles.contentInputs} pointerEvents="box-none">
        <TextToken
          theme={theme}
          index={item.index}
          value={value}
          focused={focused}
          error={error}
          maxLength={others.maxLength}
        />
        {
          isLastItem ? null : <Spacer size={theme.spacings.sNano} />
        }
      </View>
    );
  }), [listPositions, theme, value, focused, error, others]);

  const inputSelection = useMemo(() => ({
    /*
     * For√ßa o cursor para ficar sempre no FINAL do input.
     */
    start: value.length,
    end: value.length,
  }), [value.length]);

  return (
    <View testID={testID}>
      <TextInput
        {...others}
        ref={innerRef}
        testID="rn-text-input"
        caretHidden
        secureTextEntry
        value={value}
        onChangeText={onChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        style={inputStyles}
        selection={inputSelection}
      />
      <View style={contentContainerStyles} pointerEvents="box-none">
        {renderTextsToken()}
      </View>
    </View>
  );
};

InputToken.defaultProps = {
  keyboardType: 'numeric' as TextInputProps['keyboardType'],
};

InputToken.Field = withTheme(wrapField<InputTokenPropsType>((props, fieldProps) => {
  const {
    invalid, touched, value, onChange, onBlur, onFocus,
  } = fieldProps;

  const { showError } = getFieldErrorState(
    invalid,
    touched,
  );

  return (
    <InputToken
      {...props}
      value={value || ''}
      onChange={onChange}
      onBlur={onBlur}
      onFocus={onFocus}
      error={showError}
    />
  );
}));

export default withTheme(InputToken);
