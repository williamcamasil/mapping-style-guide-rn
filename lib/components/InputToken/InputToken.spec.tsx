import React from 'react';

import { render, fireEvent } from '@testing-library/react-native';
import { createForm } from 'final-form';

import InputToken from '.';
import Form from '../Form';
import { isCurrentEmptyInputFocused } from './utils';

describe('tests with InputToken', () => {
  it('should save InputToken snapshot', () => {
    const result = render(
      <InputToken
        maxLength={6}
        value=""
        error={false}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should onFocus on InputToken', () => {
    const result = render(
      <InputToken
        maxLength={6}
        value=""
        error={false}
      />,
    );

    const input = result.getByTestId('rn-text-input');
    fireEvent(input, 'focus');

    expect(result.toJSON).toMatchSnapshot();
  });

  it('should onChange on InputToken', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputToken
        maxLength={6}
        onChange={onChangeMock}
        value=""
        error={false}
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', '123');

    expect(onChangeMock).toBeCalledWith('123');
  });

  it('should onChange on InputToken and return error', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputToken
        maxLength={4}
        onChange={onChangeMock}
        value=""
        error
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', '1234');

    expect(onChangeMock).toBeCalledWith('1234');
  });

  it('should remove text indicator after removing focus from input', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputToken
        maxLength={6}
        onChange={onChangeMock}
        value="123"
        error={false}
      />,
    );

    const rnTextInput = result.getByTestId('rn-text-input');

    fireEvent(rnTextInput, 'focus');
    fireEvent(rnTextInput, 'changeText', '123');
    fireEvent(rnTextInput, 'blur');

    const textView = result.queryByText('|');

    expect(textView).toBeNull();
    expect(result.toJSON).toMatchSnapshot();
  });
});

describe('tests with InputToken Utils', () => {
  it('should call isCurrentEmptyInputFocused', () => {
    const resultFocused = isCurrentEmptyInputFocused(true, 6, 6);
    expect(resultFocused).toBeTruthy();

    const resultUnfocused = isCurrentEmptyInputFocused(false, 6, 6);
    expect(resultUnfocused).toBeFalsy();

    const resultFocusedMoreEntryNumbers = isCurrentEmptyInputFocused(false, 6, 1);
    expect(resultFocusedMoreEntryNumbers).toBeFalsy();
  });
});

describe('tests with InputToken Field', () => {
  it('should valid inputToken', async () => {
    const form = createForm({
      onSubmit: jest.fn(),
    });

    const component = render(
      <Form
        onSubmit={jest.fn()}
        form={form}
      >
        {() => (
          <InputToken.Field
            name="InputToken"
            maxLength={6}
            required
            validate={value => (value?.length === 6 ? undefined : 'Erro')}
          />
        )}
      </Form>,
    );

    expect(form.getState().valid).toBeFalsy();

    const inputSelect = component.getByTestId('rn-text-input');
    fireEvent(inputSelect, 'changeText', '123456');

    expect(form.getState().valid).toBeTruthy();
  });
});
