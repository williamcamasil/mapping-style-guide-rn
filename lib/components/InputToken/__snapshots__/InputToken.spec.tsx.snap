// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests with InputToken should onFocus on InputToken 1`] = `[Function]`;

exports[`tests with InputToken should remove text indicator after removing focus from input 1`] = `[Function]`;

exports[`tests with InputToken should save InputToken snapshot 1`] = `
<View>
  <TextInput
    caretHidden={true}
    keyboardType="numeric"
    maxLength={6}
    onBlur={[Function]}
    onFocus={[Function]}
    secureTextEntry={true}
    selection={
      {
        "end": 0,
        "start": 0,
      }
    }
    style={
      {
        "height": 64,
        "opacity": 0,
      }
    }
    testID="rn-text-input"
    value=""
  />
  <View
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-0"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-1"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-2"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-3"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-4"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "alignItems": "center",
            "borderColor": "#E1E2E4",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 64,
            "paddingTop": 16,
            "width": 44,
          }
        }
        testID="text-token-touchable-5"
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 19,
              "fontWeight": "500",
              "letterSpacing": -0.02,
              "lineHeight": 27,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
